test/language/languageUnderstandingTest.js

```diff
@@ -1,40 +1,275 @@
 const cognitive = require('../../src/index.js');
 const config = require('../config.js');
+const promiseRetry = require('promise-retry');
+const promiseDelay = require('sleep-promise');
+/*
 
-describe('Language understanding (LUIS)', () => {
+You only need to set the apiKey for these tests. 
+
+The app created is the prebuilt "Web" app. Creation of the app
+will fail if your LUIS account already has an app with the 
+name "Web". 
+
+Each time the app is created, its apiKey is displayed along
+with the count of training status calls. Usually, it takes more 
+than 1 call to return successfully trained status. This test will 
+not try more than retryCount times and wait retryInterval between tries. 
+
+*/
+
+describe.only('Language understanding (LUIS)', () => {
+
+    const defaultVersionID = "0.1";
+    const retryCount = 10;
+    const retryInterval = 1000;
+    let count = 0;
 
     const client = new cognitive.languageUnderstanding({
         apiKey: config.languageUnderstanding.apiKey,
-        appID: config.languageUnderstanding.appID,
-        versionID: config.languageUnderstanding.versionID,
         endpoint: config.languageUnderstanding.endpoint
     });
 
-    describe('Detect Intent', () => {
-        it('should return response', (done) => {
-            // optional but recommended
+    var waitUntilTrained = (client) => {
+
+        count += 1;
+
+        return promiseRetry((retry, number) => {
+
+            return promiseDelay(retryInterval)
+            .then( () => {
+                return client.getTrainStatus();
+            }).then(response => {
+                // 2xx http response 
+                let trained = client.isTrained(response);
+
+                console.log(number + " trained = " + trained);
+
+                if (count < retryCount && !trained) retry("not trained");
+                
+                return response;
+            })
+            .catch((err) => {
```

Comment:
This catch block is not necessary, if there is an error it will be caught further down the line (in the tests)